content/Payment Platforms options - Gonzalo Mercadante.md

```diff
@@ -0,0 +1,92 @@
+# Payment platforms for mobile apps
+
+Nowadays, is very common to see apps that offer you in-app content. Sometimes they are just virtual things, like coins, diamonds, credits, which give users access to certain functionalities. But also, there are a lot of e-commerce apps, like Amazon, where you have a lot of things to buy.
+Both types of apps are using payment platforms to manage the payments. And there are a lot of options you can use.
+Let’s try to list some of them, along with some pros and cons.
+
+## What payment platform should I use?
+
+Not always you can use the platform you want. If you want to sell virtual/digital goods in your app (coins, credits, etc) and you’re not providing another way to do that (for example, a web page) then you must use In-app purchases to be approved by Apple. On the other hand, if you’re implementing an e-commerce page where you’re selling for example, clothes, then you can use other payment methods.
+
+
+### Virtual goods
+
+There a lot of apps that use coins or credits to give the users access to certain functionalities, and for this case, you need to use In-app purchases. Despite the fact that most of the logic is managed in the frontend, because the payments are managed by apple/google in the device, the purchases should be validated also in the server.
+As with any feature that has anything to do with money, security is very important when implementing in-app purchases. Specifically, it’d be good to ensure that when you’re providing a user with a paid resource, they have actually paid for it.
+So, after the purchases are made, the frontend should send the receipt data to the server (encrypted data) and then the server should validate the receipt with apple/google servers.
```

Comment:
I would add that optionally, a two step validation can be implemented, where the receipt is validated in the frontend first and then validated and confirmed in the backed.

Reply:
I think the purchases should always be validated in the server since you're providing some additional features.
If you don't validate in the backend, I can call the server more than once but without receipts.

Reply:
Yes, Im not saying that the server validation should be skipped. But that there is another implementation option, that is where you validate on both ends. Just to let the reader know about the choices.

Reply:
Ohh got it, updated. Thanks!